packages/store-sync/src/react/SyncProvider.tsx

```diff
@@ -19,17 +18,18 @@ export function SyncProvider({ chainId, adapter, children, ...syncOptions }: Pro
     throw new Error("A `SyncProvider` cannot be nested inside another.");
   }
 
-  const config = useConfig();
+  const chains = useChains();
+  const chain = chains.find((c) => c.id === chainId);
+  if (!chain) {
+    throw new Error(`No chain configured for chain ID ${chainId}.`);
+  }
 
   const result = useQuery({
     queryKey: ["sync", chainId],
-    queryFn: async () => {
-      const client = getClient(config, { chainId });
-      if (!client) {
-        throw new Error(`Unable to retrieve Viem client for chain ${chainId}.`);
-      }
-      return await adapter({ publicClient: client, ...syncOptions });
-    },
+    queryFn: () =>
+      // TODO: decide if we want to pass in publicClient when not using `internal_validateBlockRange`
+      //       so that the rpc client inherits transports from wagmi config
```

Comment:
let's go back to `publicClient` by default because viem plans to add batched RPC call support in v3

only when we're inside the flag should we change to using chain and our own client